test/pipelinerun_test.go

```diff
@@ -84,6 +85,34 @@ func TestPipelineRun(t *testing.T) {
 			}
 		}
 	}
+
+	logger.Infof("Making sure events were created from taskrun and pipelinerun")
+	watchEvents, err := c.KubeClient.Kube.CoreV1().Events(namespace).Watch(metav1.ListOptions{})
+	// close watchEvents channel
+	defer watchEvents.Stop()
+	if err != nil {
+		t.Errorf("failed to create watch on events: %v", err)
+	}
+	// create timer to not wait for events longer than 5 seconds
+	timer := time.NewTimer(5 * time.Second)
+	expectedNumberOfEvents := 3
+	// 1 from PipelineRun and 2 from Tasks defined in pipelinerun
+	gotEventCount := 0
+	for {
+		select {
+		case wevent := <-watchEvents.ResultChan():
+			event := wevent.Object.(*corev1.Event)
+			// successful taskrun and pipelinerun events.
+			if event.InvolvedObject.Kind == "TaskRun" || event.InvolvedObject.Kind == "PipelineRun" && event.Reason == "Succeeded" {
+				gotEventCount++
+			}
+		case <-timer.C:
+			if gotEventCount != expectedNumberOfEvents {
+				t.Errorf("Expected %d number of successful events from pipelinerun and taskrun but got %d", expectedNumberOfEvents, gotEventCount)
+			}
+			return
+		}
+	}
```

Comment:
can we move this into a function called by the test? the test is getting pretty long and difficult to read

Reply:
and if this is in a separate function, we could call it easily from other tests

p.s. im hoping that we can sometime soon consolidate down the number of integration tests

Reply:
Yes I will extract into generic function which collects events that matches certain object and reason of event

Reply:
Thanks, looks great :D